Don't compiled cap_ffmpeg.cpp

Docker compile in Alpine 3.9

```
--
-- General configuration for OpenCV 4.1.1-pre =====================================
--   Version control:               4.1.0-545-gf663e8f90
--
--   Platform:
--     Timestamp:                   2019-07-11T09:46:30Z
--     Host:                        Linux 4.19.57-sunxi64 aarch64
--     CMake:                       3.14.5
--     CMake generator:             Unix Makefiles
--     CMake build tool:            /usr/bin/make
--     Configuration:               Release
--
--   CPU/HW features:
--     Baseline:                    NEON FP16
--       required:                  NEON
--       disabled:                  VFPV3

--   C/C++:
--     Built as dynamic libs?:      YES
--     C++ Compiler:                /usr/bin/c++  (ver 8.3.0)
--     C++ flags (Release):         -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-securi
ty -Wmissing-declarations -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Winit-self -Wno-delete-non-virtual-dtor -Wno-comment -Wimplicit-fallthro
ugh=3 -Wno-strict-overflow -fdiagnostics-show-option -pthread -fomit-frame-pointer -ffunction-sections -fdata-sections    -fvisibility=hidden -fvisibility-inlines-hidden -O3
 -DNDEBUG  -DNDEBUG

I have error:
[ 75%] Building CXX object modules/videoio/CMakeFiles/opencv_videoio.dir/src/cap_ffmpeg.cpp.o
ESC[91mIn file included from /usr/include/fortify/strings.h:22,
                 from /usr/include/string.h:57,
                 from /usr/include/fortify/wchar.h:30,
                 from /usr/include/c++/8.3.0/cwchar:44,
                 from /usr/include/c++/8.3.0/bits/postypes.h:40,
                 from /usr/include/c++/8.3.0/bits/char_traits.h:40,
                 from /usr/include/c++/8.3.0/string:40,
                 from /usr/include/c++/8.3.0/stdexcept:39,
                 from /usr/include/c++/8.3.0/array:39,
                 from /opencv/modules/core/include/opencv2/core/cvdef.h:695,
                 from /opencv/modules/core/include/opencv2/core.hpp:52,
                 from /opencv/modules/videoio/include/opencv2/videoio.hpp:46,
                 from /opencv/build/modules/videoio/precomp.hpp:45:
/usr/include/fortify/unistd.h:110:1: error: type of 'cv::read' is unknown
 _FORTIFY_FN(read) ssize_t read(int __f, void *__s, size_t __n)
 ^~~~~~~~~~~
ESC[0mESC[91mIn file included from /opencv/modules/videoio/src/cap_ffmpeg_impl.hpp:116,
                 from /opencv/modules/videoio/src/cap_ffmpeg.cpp:50:
/usr/include/fortify/unistd.h: In function 'ssize_t read(int, void*, size_t)':
/usr/include/fortify/unistd.h:116:34: error: '__orig_read' cannot be used as a function
  return __orig_read(__f, __s, __n);
                                  ^
In file included from /usr/include/fortify/strings.h:22,
                 from /usr/include/string.h:57,
                 from /usr/include/fortify/wchar.h:30,
                 from /usr/include/c++/8.3.0/cwchar:44,
                 from /usr/include/c++/8.3.0/bits/postypes.h:40,
                 from /usr/include/c++/8.3.0/bits/char_traits.h:40,
                 from /usr/include/c++/8.3.0/string:40,
                 from /usr/include/c++/8.3.0/stdexcept:39,
                 from /usr/include/c++/8.3.0/array:39,
                 from /opencv/modules/core/include/opencv2/core/cvdef.h:695,
                 from /opencv/modules/core/include/opencv2/core.hpp:52,
                 from /opencv/modules/videoio/include/opencv2/videoio.hpp:46,
                 from /opencv/build/modules/videoio/precomp.hpp:45:
/usr/include/fortify/unistd.h: At global scope:
/usr/include/fortify/unistd.h:146:1: error: type of 'cv::write' is unknown
 _FORTIFY_FN(write) ssize_t write(int __f, const void *__s, size_t __n)
 ^~~~~~~~~~~
In file included from /opencv/modules/videoio/src/cap_ffmpeg_impl.hpp:116,
                 from /opencv/modules/videoio/src/cap_ffmpeg.cpp:50:
/usr/include/fortify/unistd.h: In function 'ssize_t write(int, const void*, size_t)':
/usr/include/fortify/unistd.h:152:35: error: '__orig_write' cannot be used as a function
  return __orig_write(__f, __s, __n);
                                   ^
ESC[0m[ 75%] Building CXX object modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/padding_layer.cpp.o
[ 75%] Building CXX object modules/calib3d/CMakeFiles/opencv_calib3d.dir/src/calibration_handeye.cpp.o
ESC[91mmake[2]: *** [modules/videoio/CMakeFiles/opencv_videoio.dir/build.make:193: modules/videoio/CMakeFiles/opencv_videoio.dir/src/cap_ffmpeg.cpp.o] Error 1
ESC[0mESC[91mmake[1]: *** [CMakeFiles/Makefile2:3559: modules/videoio/CMakeFiles/opencv_videoio.dir/all] Error 2
ESC[0mESC[91mmake[1]: *** Waiting for unfinished jobs....
```

I have run into this issue as well.

```
[476/558] Building CXX object modules/videoio/CMakeFiles/opencv_videoio.dir/src/cap_ffmpeg.cpp.o
FAILED: modules/videoio/CMakeFiles/opencv_videoio.dir/src/cap_ffmpeg.cpp.o 
/usr/bin/g++  -DCVAPI_EXPORTS -DENABLE_PLUGINS -DHAVE_CAMV4L2 -DHAVE_DC1394_2 -DHAVE_FFMPEG -DHAVE_GSTREAMER -DSTRICT_PLUGIN_CHECK -D_USE_MATH_DEFINES -D__OPENCV_BUILD=1 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I. -I../modules/videoio/include -Imodules/videoio -I../modules/core/include -I../modules/imgproc/include -I../modules/imgcodecs/include -isystem /usr/include/eigen3 -isystem /usr/include/gstreamer-1.0 -isystem /usr/include/glib-2.0 -isystem /usr/lib/glib-2.0/include -isystem /usr/include/orc-0.4 -Os -fomit-frame-pointer   -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Winit-self -Wsuggest-override -Wno-delete-non-virtual-dtor -Wno-comment -Wimplicit-fallthrough=3 -Wno-strict-overflow -fdiagnostics-show-option -Wno-long-long -pthread -fomit-frame-pointer -ffunction-sections -fdata-sections  -msse -msse2 -msse3 -fvisibility=hidden -fvisibility-inlines-hidden -fopenmp -O3 -DNDEBUG  -DNDEBUG -fPIC   -std=c++11 -MD -MT modules/videoio/CMakeFiles/opencv_videoio.dir/src/cap_ffmpeg.cpp.o -MF modules/videoio/CMakeFiles/opencv_videoio.dir/src/cap_ffmpeg.cpp.o.d -o modules/videoio/CMakeFiles/opencv_videoio.dir/src/cap_ffmpeg.cpp.o -c ../modules/videoio/src/cap_ffmpeg.cpp
In file included from /usr/include/fortify/stdlib.h:32,
                 from /usr/include/c++/9.2.0/cstdlib:75,
                 from /usr/include/c++/9.2.0/stdlib.h:36,
                 from /usr/lib/gcc/x86_64-alpine-linux-musl/9.2.0/include/mm_malloc.h:27,
                 from /usr/lib/gcc/x86_64-alpine-linux-musl/9.2.0/include/xmmintrin.h:34,
                 from /usr/lib/gcc/x86_64-alpine-linux-musl/9.2.0/include/emmintrin.h:31,
                 from ../modules/core/include/opencv2/core/cv_cpu_dispatch.h:33,
                 from ../modules/core/include/opencv2/core/cvdef.h:322,
                 from ../modules/core/include/opencv2/core.hpp:52,
                 from ../modules/videoio/include/opencv2/videoio.hpp:46,
                 from ../modules/videoio/src/precomp.hpp:45,
                 from ../modules/videoio/src/cap_ffmpeg.cpp:42:
/usr/include/fortify/unistd.h:110:1: error: type of 'cv::read' is unknown
  110 | _FORTIFY_FN(read) ssize_t read(int __f, void *__s, size_t __n)
      | ^~~~~~~~~~~
In file included from ../modules/videoio/src/cap_ffmpeg_impl.hpp:116,
                 from ../modules/videoio/src/cap_ffmpeg.cpp:50:
/usr/include/fortify/unistd.h: In function 'ssize_t read(int, void*, size_t)':
/usr/include/fortify/unistd.h:116:34: error: '__orig_read' cannot be used as a function
  116 |  return __orig_read(__f, __s, __n);
      |   
```
```
In file included from /usr/include/fortify/stdlib.h:32,
                 from /usr/include/c++/9.2.0/cstdlib:75,
                 from /usr/include/c++/9.2.0/stdlib.h:36,
                 from /usr/lib/gcc/x86_64-alpine-linux-musl/9.2.0/include/mm_malloc.h:27,
                 from /usr/lib/gcc/x86_64-alpine-linux-musl/9.2.0/include/xmmintrin.h:34,
                 from /usr/lib/gcc/x86_64-alpine-linux-musl/9.2.0/include/emmintrin.h:31,
                 from ../modules/core/include/opencv2/core/cv_cpu_dispatch.h:33,
                 from ../modules/core/include/opencv2/core/cvdef.h:322,
                 from ../modules/core/include/opencv2/core.hpp:52,
                 from ../modules/videoio/include/opencv2/videoio.hpp:46,
                 from ../modules/videoio/src/precomp.hpp:45,
                 from ../modules/videoio/src/cap_ffmpeg.cpp:42:
/usr/include/fortify/unistd.h: At global scope:
/usr/include/fortify/unistd.h:146:1: error: type of 'cv::write' is unknown
  146 | _FORTIFY_FN(write) ssize_t write(int __f, const void *__s, size_t __n)
      | ^~~~~~~~~~~
In file included from ../modules/videoio/src/cap_ffmpeg_impl.hpp:116,
                 from ../modules/videoio/src/cap_ffmpeg.cpp:50:
/usr/include/fortify/unistd.h: In function 'ssize_t write(int, const void*, size_t)':
/usr/include/fortify/unistd.h:152:35: error: '__orig_write' cannot be used as a function
  152 |  return __orig_write(__f, __s, __n);
      |    
```
My guesses are it has something to do with: https://github.com/opencv/opencv/blob/master/modules/core/src/system.cpp#L1939

--- a/modules/videoio/src/precomp.hpp
+++ b/modules/videoio/src/precomp.hpp
@@ -42,6 +42,11 @@
 #ifndef __VIDEOIO_H_
 #define __VIDEOIO_H_
 
+#if defined __linux__ || defined __APPLE__ || defined __HAIKU__
+#include <unistd.h>  // -D_FORTIFY_SOURCE=2 workaround: https://github.com/opencv/opencv/issues/15020
+#endif
+
+
 #include "opencv2/videoio.hpp"
 #include "opencv2/videoio/legacy/constants_c.h"
 

